### PR TITLE
fix: ai endpoint issues and docs

### PR DESCRIPTION
**Pull request by TheSisb**
Fixes issues that became apparent after merging the previous AI endpoint PR

### CODE REVIEW

File: packages/paste-website/package.json [hunk 1]
```diff
@@ -147,6 +148,7 @@
     "markdown-to-jsx": "^7.3.2",
     "mdast-util-to-string": "^3.1.1",
     "next": "^13.1.6",
+    "pickleparser": "0.1.0",
```
Comment (TheSisb) [inline packages/paste-website/package.json:151]: required peer deps to be installed

File: internal-docs/engineering/environment-variables.md [hunk 2]
```diff
@@ -26,4 +26,6 @@ Full list of Environment variables and where they are needed.
 | CYPRESS_RECORD_KEY                 | Cypress record key so we can record our cypress runs                  | ✅    | ✅         |               |               |
 | PERCY_TOKEN                        | Percy.io token                                                        | ✅    | ✅         |               |               |
 | USE_CYPRESS_VRT                    | Key to enable the cypress vrt integration                             | ✅    |            |               |               |
-| NX_CLOUD_ACCESS_TOKEN             | Token to enable nx cloud cache for builds                             | ✅    | ✅         | ✅            | ✅            |
+| NX_CLOUD_ACCESS_TOKEN              | Token to enable nx cloud cache for builds                             | ✅    | ✅         | ✅            | ✅            |
+| OPENAI_API_KET                     | ChatGPT4 API Key                                                      | ✅    |            | ✅            |               |
```
Comment (TheSisb) [inline internal-docs/engineering/environment-variables.md:30]: I added these to netlify so they should be installed locally upon following the instructions in our docs for developing locally

File: packages/paste-website/src/pages/api/ai.ts [hunk 3]
```diff
@@ -19,8 +19,12 @@ import type {NextApiRequest, NextApiResponse} from 'next';
 export default async function handler(req: NextApiRequest, res: NextApiResponse): Promise<void> {
   const {question, secret} = req.body;
   // Exit early if the required params aren't provided
-  if (!question || secret !== process.env.OPENAI_API_SECRET) {
-    res.status(200).send({answer: 'Please provide a question'});
+  if (secret !== process.env.OPENAI_API_SECRET) {
```
Comment (TheSisb) [inline packages/paste-website/src/pages/api/ai.ts:22]: better http status code usage and splitting up these errors so it's more clear what is failing during testing

**Review by SiTaggart — DISMISSED** [review]

File: packages/paste-website/package.json (hunk 1 above)
Comment (nkrantz) [reply packages/paste-website/package.json:151]: 🥒

File: packages/paste-website/src/pages/api/ai.ts [hunk 4]
```diff
@@ -1,80 +1,224 @@
...
+      if (tokenCount >= 1500) {
+        break;
+      }
+
+      contextText += `${content.trim()}\n---\n`;
+    }
+
+    const prompt = codeBlock`
+      ${oneLine`
+        You are a very enthusiastic Supabase representative who loves
```
Comment (SiTaggart) [inline packages/paste-website/src/pages/api/ai.ts:149]: This prompt needs updating

Comment (TheSisb) [reply packages/paste-website/src/pages/api/ai.ts:149]: LOL ok

File: internal-docs/engineering/environment-variables.md [hunk 5]
```diff
@@ -26,4 +26,8 @@ Full list of Environment variables and where they are needed.
...
+| OPENAI_API_SECRET                  | To prevent unauthorized usage of OpenAI                               | ✅    |            | ✅            |               |
+| NEXT_PUBLIC_SUPABASE_URL           | Supabase db url                                                       | ✅    |            | ✅            |               |
```
Comment (SiTaggart) [inline internal-docs/engineering/environment-variables.md:32]: I don't think this needs to be a public key, can be private, and the description should be it's the supabase api url

File: netlify.toml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 [functions]
-  external_node_modules = ["@sparticuz/chromium", "puppeteer-core", "@swc"]
+  external_node_modules = ["@sparticuz/chromium", "puppeteer-core", "@swc", "faiss-node"]
```
Comment (SiTaggart) [inline netlify.toml:2]: we also don't need this dependency anymore

File: internal-docs/engineering/environment-variables.md (hunk 5 above)
Comment (SiTaggart) [reply internal-docs/engineering/environment-variables.md:32]: Infact, change it to `SUPABASE_URL`

File: internal-docs/engineering/environment-variables.md [hunk 7]
```diff
@@ -26,4 +26,8 @@ Full list of Environment variables and where they are needed.
...
+| SUPABASE_SERVICE_ROLE_KEY          | API Key to query supabase db OpenAI                                   | ✅    |            | ✅            |               |
```
Comment (SiTaggart) [inline internal-docs/engineering/environment-variables.md:33]: Change this to `SUPABASE_KEY`

And everywhere else its used.

**Review by SiTaggart — APPROVED** [review]
I resolved my own fixes 👍🏼

**Pull request merged**